### PR TITLE
Format Watcher.status.lastChecked and lastMetCondition

### LINKED ISSUE #38619: [CI] Occasional failures WatchStatusIntegrationTests#testThatStatusGetsUpdated
I got this failure locally on master, it reproduces after a few iterations with:

```
./gradlew :x-pack:plugin:watcher:unitTest -Dtests.seed=A92DF59B55106F76 -Dtests.class=org.elasticsearch.xpack.watcher.watch.WatchStatusIntegrationTests -Dtests.method="testThatStatusGetsUpdated" -Dtests.security.manager=true -Dtests.locale=en-BW -Dtests.timezone=SystemV/YST9 -Dcompiler.java=11 -Druntime.java=11
```

The error is:
```
FAILURE 4.67s | WatchStatusIntegrationTests.testThatStatusGetsUpdated <<< FAILURES!
   > Throwable #1: java.lang.AssertionError:
   > Expected: is "2019-02-08T12:25:25.22141Z"
   >      but: was "2019-02-08T12:25:25.221410Z"
   >    at __randomizedtesting.SeedInfo.seed([A92DF59B55106F76:E51670CD660E2260]:0)
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >    at org.elasticsearch.xpack.watcher.watch.WatchStatusIntegrationTests.testThatStatusGetsUpdated(WatchStatusIntegrationTests.java:49)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   >    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   >    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
   >    at java.base/java.lang.Thread.run(Thread.java:834)
```

This looks like another of the cases where the new java time formatting swallows trailing zeros. @pgomulka could you take a look, I remember reviewing https://github.com/elastic/elasticsearch/pull/38505 which looked similar.

### PR DESCRIPTION
**Pull request by pgomulka**
Change the formatting for Watcher.status.lastCheck and lastMetCondition
to be the same as Watcher.status.state.timestamp. These should all have
only millisecond precision

closes #38619

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/1

File: x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchStatusIntegrationTests.java [hunk 1]
```diff
@@ -44,10 +51,14 @@ public void testThatStatusGetsUpdated() {
         GetResponse getResponse = client().prepareGet(".watches", "doc", "_name").get();
         getResponse.getSource();
         XContentSource source = new XContentSource(getResponse.getSourceAsBytesRef(), XContentType.JSON);
+
         String lastChecked = source.getValue("status.last_checked");
+        assertThat(lastChecked, WatcherTestUtils.isSameDate(getWatchResponse.getStatus().lastChecked()));
```
Comment (jakelandis) [inline x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchStatusIntegrationTests.java]: Can you also assert that the nano seconds are not present here ?

Comment (pgomulka) [reply x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchStatusIntegrationTests.java]: sure - updated the PR

File: x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchStatusIntegrationTests.java [hunk 2]
```diff
@@ -54,11 +55,21 @@ public void testThatStatusGetsUpdated() {
...
         assertThat(lastMetCondition, is(nullValue()));
         assertThat(getWatchResponse.getStatus().lastMetCondition(), is(nullValue()));
     }
 
+    private Matcher<ZonedDateTime> isMillisResolution() {
+        return new FeatureMatcher<ZonedDateTime,Boolean>(equalTo(true), "has millisecond precision", "precission") {
+            @Override
+            protected Boolean featureValueOf(ZonedDateTime actual) {
+                //if date has millisecond precision its nanosecond field will be rounded to millis (equal millis * 10^6)
+                return actual.getNano() == actual.get(ChronoField.MILLI_OF_SECOND) * 1000_000;
```
Comment (spinscale) [inline x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchStatusIntegrationTests.java]: alternatively just `actual.getNano() % 1000 == 0` - not sure if it is more readable to be honest, so feel free to ignore me  :-)

**Review by cbuescher — APPROVED** [review]
LGTM

**Comment by gwbrown** [comment]
@pgomulka Is this still `backport pending`? It looks like it's been merged to all appropriate branches.

**Comment by pgomulka** [comment]
@gwbrown indeed not needed. Thank you for a remainder

**Pull request merged**